src/main/java/profiler/generation/R1CSConstruction.java

```diff
@@ -484,14 +519,26 @@ void constraintAssignment(ArrayList<Tuple2<Long, LinearTerm<FieldT>>> constraint
 
     // For debugging purposes
     public static <FieldT extends AbstractFieldElementExpanded<FieldT>>
-    void printMatrix(ArrayList<Tuple2<Long, FieldT>> data,
-                     long offset, long row, long col) {
+    void printMatrix(ArrayList<Tuple2<Long, FieldT>> data, long offset, long row, long col) {
         // The matrix is row x col
         for (long i = 0; i < row; i++) {
             System.out.print("Row " + i + "  ---   ");
             for (long j = 0; j < col; j++) {
                 int idx = (int) (offset + i * col + j);
-                System.out.print("[" + idx + "]" + data.get((int) idx)._2() + " ");
+                System.out.print("[" + idx + "]" + data.get(idx)._2() + " ");
+            }
+            System.out.println("");
+        }
+    }
+
+    public static <FieldT extends AbstractFieldElementExpanded<FieldT>>
+    void printSerialMatrix(LinearCombination<FieldT> data, long offset, long row, long col) {
+        // The matrix is row x col
```

Comment:
This could looks very similar to the one above. Would you be able to reuse it by calling `printMatrix(data.terms, ...)` from here?

Reply:
Yes, this makes sense. In fact I don't think this is even my function (i.e. it was already here when I arrived). Anyway, I had considered using printMatrix when first looking at this code, but couldn't figure out how at the time. Will definitely give it another look.